src/shims/tsd.h

```diff
@@ -99,6 +99,40 @@ _dispatch_thread_key_create(const unsigned long *k, void (*d)(void *))
 }
 #elif DISPATCH_USE_THREAD_LOCAL_STORAGE
 
+#if TARGET_OS_WIN32
+
+DISPATCH_TSD_INLINE
+static inline void
+_dispatch_thread_key_create(DWORD *k, void (__stdcall *d)(void *))
+{
+	dispatch_assert_zero((*k = FlsAlloc(d)));
+}
+
+struct dispatch_tsd {
+	DWORD tid;
+	void *dispatch_queue_key;
+	void *dispatch_frame_key;
+	void *dispatch_cache_key;
+	void *dispatch_context_key;
+	void *dispatch_pthread_root_queue_observer_hooks_key;
+	void *dispatch_basepri_key;
+#if DISPATCH_INTROSPECTION
+	void *dispatch_introspection_key;
+#elif DISPATCH_PERF_MON
+	void *dispatch_bcounter_key;
+#endif
+	void *dispatch_priority_key;
+	void *dispatch_r2k_key;
+	void *dispatch_wlh_key;
+	void *dispatch_voucher_key;
+	void *dispatch_deferred_items_key;
+};
+
+extern __declspec(thread) struct dispatch_tsd __dispatch_tsd;
```

Comment:
can't you try to reuse `DISPATCH_USE_THREAD_LOCAL_STORAGE` it's supposed to be the same trick, no? instead of duplicating the whole struct

Reply:
Sure, its actually the same exact structure too ... just didn't move things around to make it easier.  I can do that.